extensions.html

```diff
@@ -182,19 +184,142 @@ <h2>
+        </dt>
+        <dd>
+          <p>
+            The {{GamepadHapticActuator/canPlayEffectType()}} method steps are:
+          </p>
+          <ol>
+            <li>If this actuator can [=play effects with type=]
+            |type:GamepadHapticEffectType|, return `true`.
+            </li>
+            <li>Otherwise, return `false`.
+            </li>
+          </ol>
+        </dd>
+        <dt>
+          <dfn>playEffect()</dfn> method
+        </dt>
+        <dd>
+          <p>
+            The {{GamepadHapticActuator/playEffect()}} method steps are:
+          </p>
+          <ol>
+            <li>If |params:GamepadEffectParameters| does not describe a [=valid
+            effect=] of type |type:GamepadHapticEffectType|, return [=a promise
+            resolved with=] {{GamepadHapticsResult/"invalid-parameter"}}.
+            </li>
+            <li>Let |document| be the [=current settings object=]'s [=relevant
+            global object=]'s [=associated `Document`=].
+            </li>
+            <li>If |document| is `null` or |document| is not [=Document/fully
+            active=] or |document|'s [=visibility state=] is `"hidden"`, return
+            [=a promise resolved with=] {{GamepadHapticsResult/"preempted"}}.
+            </li>
+            <li>If [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
+            is not `undefined`:
+              <ol>
+                <li>[=Queue a global task=] on the [=relevant global object=]
+                of [=this=] using the [=gamepad task source=] to [=resolve=]
+                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
+                with {{GamepadHapticsResult/"preempted"}}.
+                </li>
+                <li>Set
+                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}} to
+                `undefined`.
+                </li>
+              </ol>
+            </li>
+            <li>If this actuator cannot [=play effects with type=] |type|,
+            return [=a promise resolved with=]
+            {{GamepadHapticsResult/"not-supported"}}.
+            </li>
+            <li>Let {{GamepadHapticActuator/[[playingEffectPromise]]}} be [=a
+            new promise=].
+            </li>
+            <li>[=Issue a haptic effect=] to the actuator with |type| and
+            |params|.
+            </li>
+            <li>Return {{GamepadHapticActuator/[[playingEffectPromise]]}}.
+            </li>
+          </ol>
+          <p>
+            When the effect completes, if it was not preempted, run these
+            steps:
+          </p>
+          <ol>
+            <li>[=Queue a global task=] on the [=relevant global object=] of
+            [=this=] using the [=gamepad task source=] to [=resolve=]
+            [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}} with
+            {{GamepadHapticsResult/"complete"}}.
+            </li>
+            <li>Set [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
```

Comment:
Clearing [[playingEffectPromise]] should be done as part of the task. Otherwise, it reads like you [[playingEffectPromise]] gets trashed before the task runs.